src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1747,11 +1812,13 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // Arguments:
     // - sender: not used
     // - e: not used
-    void TermControl::_BlinkTimerTick(const Windows::Foundation::IInspectable& /* sender */,
-                                      const Windows::Foundation::IInspectable& /* e */)
+    winrt::fire_and_forget TermControl::_BlinkTimerTick(const Windows::Foundation::IInspectable& /* sender */,
+                                                        const Windows::Foundation::IInspectable& /* e */)
     {
         if (!_IsClosing())
         {
+            co_await resume_background();
+
             _core.BlinkAttributeTick();

```

Comment:
This code is unsafe and we already have a lot of dangling pointer bugs in WT because of these things. Use `get_weak()` or make copies of WinRT objects liberally (WinRT objects are reference counted after all). 🙂

Unrelated to that, is it possible to move the blink timer into ControlCore? Having to use out-of-proc COM for blinking the cursor is kinda... weird, isn't it? 😅

Reply:
yea derp this is obviously bad. Lemme fix. 

> is it possible to move the blink timer into ControlCore in the future

Yes, we should absolutely do that. I think I started once in another branch, but that has a lot of code churn that I figured best to keep out of this discussion.